### PR TITLE
CodeMirror: EditorConfiguration.lint can be boolean

### PR DESCRIPTION
**Pull request by nnchang**

### CODE REVIEW

**Comment by vvakame** [comment]
*codemirror/codemirror.d.ts*

to author(@mihailik). could you review this PR?
:+1: or :-1:?

check list

* [ ] pass the Travic-CI test?

File: codemirror/codemirror.d.ts [hunk 1]
```diff
@@ -786,8 +786,8 @@ declare module CodeMirror {
         and thus the browser's text search works on it. This will have bad effects on performance of big documents. */
         viewportMargin?: number;
 
-        /** Optional lint configuration to be used in conjunction with CodeMirror's linter addon. */
-        lint?: LintOptions;
+        /** LintOptions|boolean. Optional lint configuration to be used in conjunction with CodeMirror's linter addon. */
+        lint?: any;
```
Comment (mihailik) [inline codemirror/codemirror.d.ts]: How about declaring it as `LintOptions|boolean` ?
The current released version of TypeScript supports that syntax.

Comment (nnchang) [inline codemirror/codemirror.d.ts]: that's fine with me, but the other places in the typing which support multiple types use 'any' as well. Union types only showed up in TS 1.4 (https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#union-types), so I assumed we're shooting for deeper backcompat. Would you prefer I change it?

Comment (vvakame) [inline codemirror/codemirror.d.ts]: please use union types. this repo target to only latest tsc.

Comment (nnchang) [inline codemirror/codemirror.d.ts]: right on; done

**Comment by vvakame** [comment]
@mihailik thanks!
@nnchang thanks mate!

**Pull request merged**